src/middleware/graphql/graphqlish.service.js

```diff
@@ -0,0 +1,161 @@
+const { parse } = require('graphql');
+const { search } = require('../../services/base/base.service');
+const { getCurrentDate, mapEOB } = require('./graphqlish.mappers');
+
+function filter(eobData, billablePeriod, allowedAmount, youPaid, insurancePaid, claimTypes) {
+  let filteredData = eobData.entry;
+  filteredData =
+    filteredData === null || filteredData === void 0
+      ? void 0
+      : filteredData.filter((bundleEntry) => {
+          var _a, _b;
+          let claimTypeMatches = true;
+          if (claimTypes) {
+            const claimTypesMapping = {
+              medical: ['institutional', 'professional'],
+              prescription: ['pharmacy'],
+              dental: ['oral'],
+              vision: ['vision'],
+              lab: [],
+            };
+            const claimTypesToBeMatched = [];
+            claimTypes.forEach((claimType) =>
+              // eslint-disable-next-line security/detect-object-injection
+              claimTypesToBeMatched.push(...claimTypesMapping[claimType])
+            );
+            claimTypeMatches = bundleEntry.resource.type.coding.some(
+              (coding) => claimTypesToBeMatched.indexOf(coding.code) !== -1
+            );
+          }
+          if (!claimTypeMatches) {
+            return false;
+          }
+          let billablePeriodMatches = true;
+          if (billablePeriod) {
```

Comment:
Probably for later, would it be worth refactoring these IF conditionals out to its own methods so it becomes more testable and reduce the nested IFs?